core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -129,7 +133,9 @@ public void close() {
       HEARTBEAT_CANCEL_POOL.submit(new Runnable() {
         @Override
         public void run() {
+          NUM_PENDING_HEARTBEAT_CLOSE.incrementAndGet();
           mHeartbeat.cancel(true);
+          NUM_PENDING_HEARTBEAT_CLOSE.decrementAndGet();
```

Comment:
yeah, that is what I tried initially but I wasn't able to get it working right using Whitebox (setInternalState). so i switched to this I will look into that later.